tools/src/launcher/Main.hs

```diff
@@ -91,90 +94,44 @@ data LauncherOptions = LO
-        help    "Path to log config that will be used by the node." <>
-        metavar "PATH"
-    loNodeLogPath <- optional $ textOption $
-        long    "node-log-path" <>
-        help    ("File where node stdout/err will be redirected " <>
-                 "(def: temp file).") <>
-        metavar "PATH"
+data LauncherConfigParseError = LauncherConfigParseError !FilePath !(Y.ParseException)
+    deriving (Show)
 
-    -- Wallet-related args
-    loWalletPath <- optional $ textOption $
-        long    "wallet" <>
-        help    "Path to the wallet frontend executable (e. g. Daedalus)." <>
-        metavar "PATH"
-    loWalletArgs <- many $ textOption $
-        short   'w' <>
-        help    "An argument to be passed to the wallet frontend executable." <>
-        metavar "ARG"
-
-    -- Update-related args
-    loUpdaterPath <- textOption $
-        long    "updater" <>
-        help    "Path to the updater executable." <>
-        metavar "PATH"
-    loUpdaterArgs <- many $ textOption $
-        short   'u' <>
-        help    "An argument to be passed to the updater." <>
-        metavar "ARG"
-    loUpdateArchive <- optional $ textOption $
-        long    "update-archive" <>
-        help    "Path to the update archive, it will be passed to the updater." <>
-        metavar "PATH"
-    loUpdateWindowsRunner <- optional $ textOption $
-        long    "updater-windows-runner" <>
-        help    "Path to write the Windows batch file executing updater" <>
-        metavar "PATH"
+instance Exception LauncherConfigParseError
 
-    -- Other args
-    loNodeTimeoutSec <- option auto $
-        long    "node-timeout" <>
-        help    ("How much to wait for the node to exit before killing it " <>
-                 "(and then how much to wait after that).") <>
-        metavar "SEC"
-    loReportServer <- optional $ strOption $
-        long    "report-server" <>
-        help    "Where to send logs in case of failure." <>
-        metavar "URL"
-    loLauncherLogsPrefix <- optional $ strOption $
-        long    "launcher-logs-prefix" <>
-        help    "Where to put launcher logs (def: console only)." <>
-        metavar "DIR"
-
-    loConfiguration <- configurationOptionsParser
-
-    pure LO{..}
+configPathParser :: Parser ConfigPath
+configPathParser = do
+    configPath <- strOption $
+        short   'c' <>
+        long    "config" <>
+        help    "Path to the launcher configuration file." <>
+        metavar "PATH"
+    pure $ ConfigPath configPath
 
 getLauncherOptions :: IO LauncherOptions
-getLauncherOptions = execParser programInfo
+getLauncherOptions = do
+    configPath <- unConfigPath <$> execParser programInfo
+    decoded <- Y.decodeFileEither configPath
+    case decoded of
+        Left err -> do
+            writeFile "config-error.log" $
+                sformat ("Failed to parse "%build%": "%shown) configPath err
+            throwIO $ LauncherConfigParseError configPath err
```

Comment:
According to the new, not yet approved, [guidelines](https://github.com/input-output-hk/cardano-sl/blob/edcbb8d3bc2659692e8145710109a1dd43f99164/docs/exceptions.md#impure-code-regular-errors) (if the link becomes obsolete, see #2059), you should use `throwM` rather than `throwIO`.

For now it doesn't matter because those guidelines are not approved, but if we force everybody to follow them before your PR is merged, this line can lead to rejection of this PR :)